integration_tests.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh -e
+
+printUsage() {
+   echo "Usage : "
+   echo "./integration_tests.sh URL BRANCH JAMES_IP JAMES_IMAP_PORT"
+   echo "    URL : URL of the repository to pull"
+   echo "    BRANCH : Branch to build"
+   echo "    JAMES_IP : IP of the James server to be tests"
+   echo "    JAMES_IMAP_PORT : Exposed IMAP port of this James server"
+   exit 1
+}
+
+if [ "$#" -ne 4 ]; then
+    printUsage
+fi
+
+URL=$1
+BRANCH=$2
+
+export JAMES_ADDRESS=$3
+export JAMES_IMAP_PORT=$4
+
+git clone $URL -b $BRANCH
+cd james-parent
+git submodule init
+git submodule update
+
+mvn -Dtest=ExternalJamesTest -DfailIfNoTests=false test
```

Comment:
Ok, then don't invoque it on the top level, use -am and -pl options.

On Wed, Jul 15, 2015 at 3:56 PM, Tellier Benoit <notifications@github.com>
wrote:

> In integration_tests.sh
> <https://github.com/linagora/james-parent/pull/4#discussion_r34680064>:
>
> > +if [ "$#" -ne 4 ]; then
> > +    printUsage
> > +fi
> > +
> > +URL=$1
> > +BRANCH=$2
> > +
> > +export JAMES_ADDRESS=$3
> > +export JAMES_IMAP_PORT=$4
> > +
> > +git clone $URL -b $BRANCH
> > +cd james-parent
> > +git submodule init
> > +git submodule update
> > +
> > +mvn -Dtest=ExternalJamesTest -DfailIfNoTests=false test
>
> Simply because maven failed by default on subproject with no tests ( api*
> for instance )
>
> Adding this option solved the problem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/linagora/james-parent/pull/4/files#r34680064>.
>